### PR TITLE
Add an FAQ.md doc

### PR DESCRIPTION
**Pull request by jeffmo**
Various FAQ's pop up on esdiscuss and elsewhere from time to time, so I figured this would be a good defacto place to log and point anyone who has FAQs.

### CODE REVIEW

**Comment by bterlson** [comment]
This seems like a pretty good start to me, although I'd just remove the todo item for destructuring until someone wants to author it.

File: FAQ.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+The motivation for `=>` was to address the oft-fired footgun of dynamic `this` bindings. Additionally, having two forms of arrows is confusing; So only one form was added.
+
+### Destructuring
+
+#####Why isn't the object property destructuring syntax flipped the other way?
+
+(i.e. `let { x: y }` vs `let { y: x }`)
+
+\<\<TODO\>\>
```
Comment (jeffmo) [inline FAQ.md]: Doh, didn't meant to leave this as a TODO.

...but...does anyone have a meaningful answer for this? (cc @dherman)

File: FAQ.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+#Frequently Asked Questions
```
Comment (UltCombo) [inline FAQ.md]: Don't you need to add spaces between headings' `#` and text for GitHub to render the heading properly?

Comment (bterlson) [inline FAQ.md]: ###Valid Feedback
####Headers do not render properly without spaces

Comment (jeffmo) [inline FAQ.md]: Seems to work fine here: https://github.com/jeffmo/ecma262/blob/master/FAQ.md
Will add spaces if you prefer, though

**Comment by jeffmo** [comment]
Updated the destructuring TODO with an answer

**Comment by getify** [comment]
I think this FAQ is a good idea, but I'd recommend annotating the "answers" with links to relevant es-discuss threads and/or TC39 notes. The reason is, as it stands, these answers could be inviting more discussion (aka "bikeshedding") as they're just quick summaries.

For example, a drive-by reader may read the discussion on `->` and find it unsatisfying, and immediately jump to more bikeshedding on it. A way to combat that is to have links in that section to the myriad places this was discussed in the past, as justification for the end result summary presented.

**Comment by jeffmo** [comment]
@getify: Yea, adding links to relevant esdiscuss threads is probably a good idea. If you have any for any of these answers feel free to comment with a link (or send a PR) and I'd be happy to include them!.

I'm also not averse to clarifying any answers now and in the future there are bits that are missing info/context/etc.

**Comment by bterlson** [comment]
I think this is good for now. Will of course accept any subsequent PRs adding context and other clarifying information.

File: FAQ.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+An index of frequently asked questions regarding all things ECMA-262.
+
+# Process Questions
+
+##### What is the process for proposing a new feature?
+
+New features start life as a proposal to the [TC39](#what-is-a-tc39) committee and must be championed (or co-championed) by at least one member of the committee. Once the proposal is raised at a committee meeting, it will become a Stage 0 proposal and move along from there. For more details on how proposal stages work, check out the [proposal process document][proposal-process-document].
+
+If you would like to contribute, please check out [Contributing to ECMAScript](https://github.com/jeffmo/ecma262/blob/master/CONTRIBUTING.md).
```
Comment (hzoo) [inline FAQ.md]: Change `jeffmo` to `tc39` in `https://github.com/jeffmo/ecma262/blob/master/CONTRIBUTING.md` or use the local path if possible (`CONTRIBUTING.md`)? :smile:

Comment (bterlson) [inline FAQ.md]: ha, good catch, I'll fix this now.

File: FAQ.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    return false;
+  } else {
+    return isEven(num - 1);
+  }
+}
+```
+
+```javascript
+// main.js
+import {isOdd} from "./OddNumberUtils";
```
Comment (hzoo) [inline FAQ.md]: Should `OddNumberUtils` be `./Odd`?

File: FAQ.md (hunk 3 above)
Comment (bterlson) [inline FAQ.md]: 4269fc4

Comment (jeffmo) [inline FAQ.md]: Damn! My plot to poison the contribution process has been foiled...

File: FAQ.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+### Destructuring
+
+##### Why isn't the object property destructuring syntax flipped the other way?
+
+(i.e. `let {x: y} = {x: 42}` vs `let {y: x} = {x: 42}`)
+
+In all other object patterns in the language, the syntax to the left of the colon represents the "structure" of an object; So having destructuring patterns match this convention was most consistent.
+
+More fundamentally, however, flipping the syntax the other way would produce a grammar that requires infinite lookahead to properly disambiguate.
```
Comment (rossberg) [inline FAQ.md]: This is an odd answer. The real reason is that the whole idea of destructuring is to use _patterns_ that ought to mirror the structure of a value -- intuitively, the value _matches_ the pattern, filling in variables. That is in line with other languages that have similar features. Clearly, `let {[a,b]: x} = {x: [1,2]}` would not make sense to anyone.

Comment (jeffmo) [inline FAQ.md]: Is that not almost exactly what I said?

> In all other object patterns in the language, the syntax to the left of the colon represents the "structure" of an object; So having destructuring patterns match this convention was most consistent.

Comment (rossberg) [inline FAQ.md]: Yea, I suppose. But I was a bit thrown off by the "More fundamentally" in the second paragraph, which never even was a relevant consideration in this case.

**Pull request merged**